packages/schema-generator/src/generate_schema.ts

```diff
@@ -0,0 +1,100 @@
+import { TypescriptDataSchemaGenerator } from '@aws-amplify/graphql-schema-generator';
+import * as fs from 'fs';
+import { AmplifyUserError } from '@aws-amplify/platform-core';
+
+export type SchemaGeneratorConfig = {
+  connectionString: string;
+  out: string;
+};
+
+/**
+ * Schema generator class.
+ */
+export class SchemaGenerator {
+  generate = async (props: SchemaGeneratorConfig) => {
+    const dbConfig = parseDatabaseUrl(props.connectionString);
+
+    try {
+      const schema = await TypescriptDataSchemaGenerator.generate(dbConfig);
+      fs.writeFileSync(props.out, schema);
+    } catch (err) {
+      const databaseError = err as DatabaseConnectError;
```

Comment:
err could be other things as well

Reply:
I have made the fields optional. In case of other errors, it will throw the error from line 37.

Reply:
it would be safer to check for `err instanceof DatabaseConnectError`